src/RankMovie.jsx

```diff
@@ -69,39 +68,105 @@ function RankMovie() {
   }
 
   const handleComparison = async (prefersNewMovie) => {
-    const winner = prefersNewMovie ? newUserMovie : userMovies[mid];
-    const loser = prefersNewMovie ? userMovies[mid] : newUserMovie;
+    //get newly logged movie, initial
+    if (!currentUser || !newMovie || mid === undefined) {
+      console.error("Current user or new movie data is not available.");
```

Comment:
Make sure not to use `console.log` for messages to the user. You can use it during development just remove them before checking in the code.